crates/ruff_linter/src/fix/snippet.rs

```diff
@@ -35,6 +35,11 @@ impl SourceCodeSnippet {
     /// Returns `true` if the source code should be truncated when included in a user-facing
     /// diagnostic.
     fn should_truncate(source_code: &str) -> bool {
-        source_code.width() > 50 || source_code.contains(['\r', '\n'])
+        source_code
+            .chars()
+            .map(|c| c.width().unwrap_or(0))
+            .sum::<usize>()
+            > 50
```

Comment:
I think we can keep using `width` here as it isn't related to any "overlong" logic.